### ISSUE TITLE
Now working in combination with barba.js

### ISSUE THREAD

**Issue by kratos-digital** [Gyanreyer/hover-video-player#5]
I am using WP for one of my projects and barba.js to make it SPA-like.

Plugin is working fine without barba.js and on first page load, however it doesn't work when navigating through pages (using barba) and I can't find any functions to be used to reinit/reload elements.

I tried setting manual hoverTarget which triggers on each page load, but this also works only on first load:

```
const players = document.querySelectorAll("hover-video-player");
players.forEach((player) => {
  console.log("set hover target", player);
  player.hoverTarget = document.querySelectorAll(".some-hover-target");
});
```

Output in log is exactly the same for first load and on each page load (through barba), but hover effect is not working.

Is there function I can call to reinit/reload plugin?

**Comment by kratos-digital** [comment]
I managed to make it work like this, but I still would like to know if there is a way to do it the default way using custom element:

```
import HoverVideoPlayer from "hover-video-player";

const videos = document.querySelectorAll("[data-video]");
videos.forEach((el) => {
  let player = new HoverVideoPlayer();
  player.hoverTarget = el;
  player.playbackStartDelay = 250;
  player.classList.add("w-full");
  player.appendChild(el.querySelector("video"));
  el.appendChild(player);
});
```

The `[data-video]` element is just `figure` element that wraps my video, instead of custom `hover-video-player` element

**Comment by Gyanreyer** [comment]
Good question! Something definitely doesn't seem to be working right here. Would you happen to have a codepen or anything you could share to help reproduce the issue?

**Comment by kratos-digital** [comment]
Wordpress on codepen? I used codepen for simple stuff, I don't know if this is possible.
I could send you zip of test project using barba and your plugin for testing purposes, but will need some contact info where to send it.

**Comment by Gyanreyer** [comment]
Hi, I was able to put together a simple reproduction of the issue. It looks like this is a fundamental problem with the barba.js library where it breaks web components; for whatever reason, any custom elements like `<hover-video-player>` don't get properly connected after a barba transition. Unfortunately this means there aren't really any changes I can offer on this library's end to fix the issue, but this is the simplest implementation I have found for dealing with it:

```js
barba.hooks.enter((data) => {
  data.next.container
    .querySelectorAll("hover-video-player")
    .forEach((el) => {
      el.replaceWith(el.cloneNode(true));
    });
});
```

This will grab any new incoming `hover-video-player` elements from a transition and will clone and re-mount them all.

I hope that helps!
